internal/file/file.go

```diff
@@ -43,11 +44,14 @@ func HasExtension(path string, extensions []string) bool {
 }
 
 // ShouldIgnore returns true if the path matches any of the ignore strings
-func ShouldIgnore(path string, ignoreFolders []string) bool {
+func ShouldIgnore(path string, ignores []string) bool {
 	segments := strings.Split(path, string(os.PathSeparator))
-	for _, segment := range segments {
-		for _, ignore := range ignoreFolders {
-			if segment == ignore {
+	for _, ignore := range ignores {
+		sub := strings.Split(ignore, string(os.PathSeparator))
+		size := len(sub)
+		last := len(segments) - size
+		for i := 0; i <= last; i++ {
+			if reflect.DeepEqual(segments[i:i+size], sub) {
```

Comment:
I suggest to be more descriptive (think in your future you): 
segments -> pathSegments
ignores -> pathsToIgnore
ignore -> pathToIgnore
sub -> ignorePathSegments
...

Reply:
yep